mriqc/workflows/diffusion/base.py

```diff
@@ -416,6 +413,80 @@ def dmri_bmsk_workflow(name="dmri_brainmask", omp_nthreads=None):
     return workflow
 
 
+def init_dmriref_wf(
+    in_file=None,
+    multiecho=False,
+    name="init_dmriref_wf",
+):
+    """
+    Build a workflow that generates reference images for a dMRI series.
+
+    The raw reference image is the target of :abbr:`HMC (head motion correction)`, and a
+    contrast-enhanced reference is the subject of distortion correction, as well as
+    boundary-based registration to T1w and template spaces.
+
+    This workflow assumes only one dMRI file has been passed.
+
+    Workflow Graph
+        .. workflow::
+            :graph2use: orig
+            :simple_form: yes
+
+            from mriqc.workflows.diffusion.base import init_dmriref_wf
+            wf = init_dmriref_wf()
+
+    Parameters
+    ----------
+    in_file : :obj:`str`
+        dMRI series NIfTI file
+    ------
+    in_file : str
+        series NIfTI file
+
+    Outputs
+    -------
+    in_file : str
+        Validated DWI series NIfTI file
+    ref_file : str
+        Reference image to which DWI series is motion corrected
+    """
+    from niworkflows.interfaces.images import RobustAverage
+    from niworkflows.interfaces.header import ValidateImage
+
+    workflow = pe.Workflow(name=name)
+    inputnode = pe.Node(niu.IdentityInterface(fields=["in_file"]), name="inputnode")
+    outputnode = pe.Node(
+        niu.IdentityInterface(fields=["in_file", "ref_file", "validation_report"]),
+        name="outputnode",
+    )
+
+    # Simplify manually setting input image
+    if in_file is not None:
+        niu.IdentityInterface(fields=["in_file"]),
+
+    val_bold = pe.Node(
+        ValidateImage(),
+        name="val_bold",
+        mem_gb=DEFAULT_MEMORY_MIN_GB,
+    )
+
+    gen_avg = pe.Node(RobustAverage(), name="gen_avg", mem_gb=1)
+
+    # fmt: off
+    workflow.connect([
+        (inputnode, val_bold, [("in_file", "in_file")]),
+        (val_bold, gen_avg, [("out_file", "in_file")]),
```

Comment:
Inform `RobustAverage` of where the b=0 are in the series:
```suggestion
        (val_bold, gen_avg, [("out_file", "in_file"),
                             ("t_mask", "t_mask"),
        ]),
```

Reply:
I think instead you want:

```suggestion
        (inputnode, gen_avg, [("t_mask", "t_mask")]),
        (val_bold, gen_avg, [("out_file", "in_file")]),
```

Reply:
Correct -- thanks for catching this